client/signup/steps/business-details/index.tsx

```diff
@@ -0,0 +1,221 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import { get } from 'lodash';
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { localize, LocalizeProps } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import Button from 'components/button';
+import Card from 'components/card';
+import FormFieldset from 'components/forms/form-fieldset';
+import FormTextInput from 'components/forms/form-text-input';
+import StepWrapper from 'signup/step-wrapper';
+import { getBusinessAddress, getBusinessName } from 'state/signup/steps/business-details/selectors';
+import { setBusinessAddress, setBusinessName } from 'state/signup/steps/business-details/actions';
+import { recordTracksEvent } from 'state/analytics/actions';
+import { saveSignupStep, submitSignupStep } from 'state/signup/progress/actions';
+import wpcom from 'lib/wp';
+
+/**
+ * Style dependencies
+ */
+import './style.scss';
+
+interface Props {
+	flowName: string;
```

Comment:
Do we have a known set of `flowNames`? Would there be benefit to expressing the type of known `flowName`s through the type system?

Reply:
Yes, good point. All valid flow names are defined in https://github.com/Automattic/wp-calypso/blob/master/client/signup/config/flows-pure.js. I think it would be beneficial to limit this type to defined flows.